### ISSUE TITLE
Optionally empty tag on empty string?

### ISSUE THREAD

**Issue by bil-bas** [jgeewax/pyhaml-jinja#3]
e.g. assuming frog is False

```
%a(tag="{{ 'fish' if frog else '' }}")  => <a tag></a>
```

Shouldn't this give you "<a></a>" without a tag?

**Comment by jgeewax** [comment]
That's a good point. What does regular HAML do here?

I suspect that we should distinguish between `None` and `''`, perhaps `None` should remove the tag entirely, and `''` would do as you see above.

**Comment by bil-bas** [comment]
I'd like to see None working that way, but not possible the way jinja works (it converts the None to a string at run-time, not when you parse):

```
%a(tag="{{ 'fish' if frog else None }}")  => <a tag="None"></a>
```

**Comment by jgeewax** [comment]
I believe we're doing some pre-processing here, so I'll at least give it a try.

**Comment by bil-bas** [comment]
You could fiddle it to recognise None, but what if we have "{{ frog }}" and frog just happens to be None?

**Comment by jgeewax** [comment]
I would expect that `%a(tag="{{ None }}")` would translate to `<a></a>`... no?

More specifically, the same result should come from:

```
-set frog = None
%a(tag="{{ frog }}")
```

(`<a></a>`)

**Comment by bil-bas** [comment]
No. the {{ }} convert the result into a string and then you'd have tag="None"

**Comment by bil-bas** [comment]
That is, convert `%a(tag={{ None }})`  to jinja template `<a tag="{{ None }}"></a>` which is then converted to HTML `<a tag="None"></a>`

**Comment by jgeewax** [comment]
Sorry, let me be more explicit.

Right now, `%a(tag="{{ expr }}")` translates into `<a tag="{{ expr }}"></a>`.

I'm suggesting that we consider translating it to:

```
<a
{%- if expr is not None %} tag="{{ expr }}"{% endif -%}
></a>
```

**Comment by bil-bas** [comment]
Oh, I see. That seems a bit of a hack really and may not work as people expect.

What if the expression is not consistent e.g. `(get_next_value_in_queue()`). A better, though still horrible way would be (remembering to put braces around the expression as you would with C macros :D ):

{% __exp_result = (expr) %}
{%- if __exp_result is not None %} tag="{{ __exp_result }}"{% endif -%}

**Comment by jgeewax** [comment]
That's a great point -- still do-able but definitely tricky.

Maybe we should leave as is and ask the user to do this? (ie, if you want empty string to get rid of the property all together, just wrap in the if-statement?)

**Comment by bil-bas** [comment]
Yeah, I suppose so. Can always just write it all in regular HTML if it matters enough.

Still not sure about `frog=""` being `frog` though ;) Can't remember what it does in "real" HAML
